### PR TITLE
18954: Axises titles incorrectly located, rotate property does not work

### PR DESCRIPTION
**Pull request by savvinsergey**

### CODE REVIEW

File: src/visual.ts [hunk 1]
```diff
@@ -123,14 +123,16 @@ module powerbi.extensibility.visual {
...
+        private static TickHeight: number = 6;
         private static EmptyDisplayName: string = "";
         private static MinLabelSize: number = 0;
         private static MiddleOfTheLabel: number = 2;
 
-
         private static DefaultDataLabelsOffset: number = 4;
         private static DefaultLabelTickWidth: number = 10;
 
+        private static Sin60: number = 0.8660266281835431;
```
Comment (ignatvilesov) [inline src/visual.ts]: Are you sure about numbers? Can we compute this in the run time?

File: src/visual.ts [hunk 2]
```diff
@@ -123,14 +123,16 @@ module powerbi.extensibility.visual {
...
+        private static Cos60: number = 0.5000001943375613;
```
Comment (ignatvilesov) [inline src/visual.ts]: Are you sure about numbers? Can we compute this in the run time?

File: src/visual.ts [hunk 3]
```diff
@@ -477,12 +490,37 @@ module powerbi.extensibility.visual {
 
             this.renderLegend(this.data);
 
-
             this.svg.attr({
                 "width": PixelConverter.toString(this.viewport.width),
                 "height": PixelConverter.toString(this.viewport.height)
             });
 
+            this.calculateAxes();
```
Comment (ignatvilesov) [inline src/visual.ts]: It'd be better to move all axis related code to a new method

File: src/visual.ts [hunk 4]
```diff
@@ -477,12 +490,37 @@ module powerbi.extensibility.visual {
...
+
+            this.renderXAxisLabels();
+            this.renderYAxisLabels();
+
+            this.axes.attr("transform", SVGUtil.translate(this.margin.left, 0));
+            this.axisX.attr("transform", SVGUtil.translate(0, this.viewport.height - this.margin.bottom));
+            this.axisY.attr("transform", SVGUtil.translate(0, this.margin.top));
+
+            if (this.data.settings.categoryAxis.show) {
```
Comment (ignatvilesov) [inline src/visual.ts]: These two conditions are the same. Please create a common method instead.

File: src/visual.ts [hunk 5]
```diff
@@ -513,42 +551,40 @@ module powerbi.extensibility.visual {
...
 
-            this.renderXAxisLabels();
-            this.renderYAxisLabels();
+            const rotatedTextWidth: number = StreamGraph.XAxisLabelSize * StreamGraph.Sin60,
+                rotatedLongestTextHeight: number = this.data.maxNumberOfAxisXValues * StreamGraph.Cos60;
+
+            return rotatedLongestTextHeight + rotatedTextWidth + StreamGraph.TickHeight;
+        }
+
+        private setTextNodesPosition(xAxisTextNodes, textAnchor, dx, dy, transform) {
```
Comment (ignatvilesov) [inline src/visual.ts]: Please specify type for each argument

File: src/visual.ts (hunk 1 above)
Comment (savvinsergey) [inline src/visual.ts]: Yes, i am sure. This values have already used

File: src/visual.ts (hunk 2 above)
Comment (savvinsergey) [inline src/visual.ts]: Yes, i am sure. This values have already used

File: src/visual.ts (hunk 4 above)
Comment (savvinsergey) [inline src/visual.ts]: Done

File: src/visual.ts (hunk 3 above)
Comment (savvinsergey) [inline src/visual.ts]: Done

File: src/visual.ts (hunk 5 above)
Comment (savvinsergey) [inline src/visual.ts]: Done

File: src/visual.ts [hunk 6]
```diff
@@ -513,42 +527,57 @@ module powerbi.extensibility.visual {
...
+            xAxisTextNodes
+                .style("text-anchor", textAnchor)
+                .attr({ dx, dy, transform });
+        }
+
+        private toggleAxisVisibility(
+            isShow: boolean,
+            axisType: string): void {
+
+            this["axis" + axisType].classed(StreamGraph[axisType + "Axis"].class, isShow);
```
Comment (ignatvilesov) [inline src/visual.ts]: It'd be better to use an object directly and specify it in the arguments of the method instead.
```javascript
toggleAxisVisibility(isShown: boolean, axis: d3.svg.axis) {
  axis.classed(...
```

Comment (savvinsergey) [inline src/visual.ts]: Done

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,4 +1,6 @@
 ## 1.0.1
+* Fixed incorrectly location of axises titles
```
Comment (ignatvilesov) [inline CHANGELOG.md]: 1.0.1 was released yesterday. Please don't include any extra release notes. Please update version of the visual if you want to describe release notes.

Comment (savvinsergey) [inline CHANGELOG.md]: Done

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,4 +1,6 @@
-## 1.0.1
+## 1.2.0
```
Comment (ignatvilesov) [inline CHANGELOG.md]: Why did you remove 1.0.1 notes?

File: CHANGELOG.md [hunk 9]
```diff
@@ -1,3 +1,8 @@
+## 1.1.0
+* Fixed incorrectly location of axises titles
+* Fixed rotate property does not work
+* Fixe x-axis labels overlaps
```
Comment (ignatvilesov) [inline CHANGELOG.md]: `Fixed x-axis labels overlaps`

Comment (savvinsergey) [inline CHANGELOG.md]: Done

File: CHANGELOG.md (hunk 8 above)
Comment (savvinsergey) [inline CHANGELOG.md]: Returned
